sync-core/src/main/java/com/cloudant/sync/datastore/DatastoreManager.java

```diff
@@ -153,8 +156,8 @@ public String getPath() {
     public Datastore openDatastore(String dbName) throws DatastoreNotCreatedException {
         Preconditions.checkArgument(dbName.matches(LEGAL_CHARACTERS),
```

Comment:
This should drop down into `openDatastore(Stirng dbName, KeyProvider provider)` providing a suitable default for the key provider